src/meltano/core/plugin/project_plugin.py

```diff
@@ -370,23 +398,30 @@ def venv_name(self) -> str:
 
         return self.name
 
+    @property
+    def all_requires(self) -> dict[PluginType, list]:
+        """Return all requires for this plugin.
+
+        Returns:
+            List of supported requires, including those inherited from the parent plugin.
+        """
+        return {
+            plugin_type: [
+                *self._parent.all_requires.get(plugin_type, []),
+                *self.requires.get(plugin_type, []),
+            ]
+            for plugin_type in list(PluginType)
+        }
+
     @property
     def requirements(self) -> list[ProjectPlugin]:
         """Return the requirements for this plugin.
 
         Returns:
             A list of requirements for this plugin.
         """
-        plugins = [
+        return [
             ProjectPlugin(plugin_type=plugin_type, name=dep.name, variant=dep.variant)
-            for plugin_type, deps in self.requires.items()
+            for plugin_type, deps in self.all_requires.items()
             for dep in deps
         ]
-
-        plugins.extend(
-            ProjectPlugin(plugin_type=plugin_type, name=dep.name, variant=dep.variant)
-            for plugin_type, deps in self._parent.requires.items()
```

Comment:
We weren't correctly inheriting multiple layers of `requires` before, only the direct parents, but not the parent's parents('s parents).

Reply:
can we construct a test case for this to make sure we don't drift again ?

Reply:
@pandemicsyn That might be addressed in https://github.com/meltano/meltano/pull/5999/ but I forgot we hand0t merged that 🤦

Reply:
oh heh , yea - y'all are definitely gonna have a conflict 😁 @edgarrmondragon do you wanna just let this change roll first and rebuild on top of it?

Reply:
@pandemicsyn I can wait for this to merge, but I'm not sure if we're blocking @pnadolny13

Reply:
@edgarrmondragon no its not blocking me, I just needed it fixed in order to confirm that my changes in https://github.com/meltano/hub/pull/538 were working. Do whatever is best for you all.